### PR TITLE
Decouple RPC calls from the VerifyingProvider 

### PR DESCRIPTION
**Pull request by shresthagrawal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
     "@ethereumjs/tx": "4.0.0",
     "@ethereumjs/util": "8.0.0",
     "@ethereumjs/vm": "6.0.0",
+    "@types/axios": "0.14.0",
```
Comment (pkakelas) [inline package.json:26]: Let's move that to the devDependencies

File: src/rpc.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        const r = await this._retryRequest(request);
+        res.push(r);
+      }
+      return res;
+    }
+  }
+
+  private async _retryRequest(
+    _request: RPCRequest,
+    retry: number = 5,
```
Comment (pkakelas) [inline src/rpc.ts:71]: Since you have a default parameter type inference will work here. So feel free to omit the type.

File: src/rpc.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        ..._request,
+        jsonrpc: '2.0',
+        id: Math.floor(Math.random() * 2 ** 64).toFixed(),
+      },
+    ];
+
+    for(let i = retry; i > 0; i--) {
+      const res = await this._request(request);
+      if (res[0].success) return res[0];
+      else if(i == 1) {
```
Comment (pkakelas) [inline src/rpc.ts:84]: Space between `if` and `(`. Also, I'd omit the else statement.

File: src/rpc.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  ): Promise<RPCResponse> {
+    const request = [
+      {
+        ..._request,
+        jsonrpc: '2.0',
+        id: Math.floor(Math.random() * 2 ** 64).toFixed(),
+      },
+    ];
+
+    for(let i = retry; i > 0; i--) {
```
Comment (pkakelas) [inline src/rpc.ts:81]: Space between `for` and `(`. Also let's make this `for` consistent with one on line 109.

File: src/rpc.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  private async _retryBatch(
+    _requests: RPCRequest[],
+    retry: number = 5,
+  ): Promise<RPCResponse[]> {
+    let requestsRaw: RPCRequestRaw[] = _requests.map(r => ({
+      ...r,
+      jsonrpc: '2.0',
+      id: Math.floor(Math.random() * 2 ** 64).toFixed(),
```
Comment (pkakelas) [inline src/rpc.ts:104]: I would decouple the ID generation from the actual request generation.

**Review by pkakelas — APPROVED** [review]

**Comment by shresthagrawal** [comment]
@pkakelas thanks for all the comments. I have fixed them

**Pull request merged**